Gruntfile.js

```diff
@@ -218,7 +218,7 @@ const configureGrunt = function (grunt) {
                         git submodule sync
                         git submodule update
 
-                        if ! git diff --exit-code --quiet; then
+                        if ! git diff --exit-code --quiet --ignore-submodules=untracked; then
```

Comment:
This means that a submodule will not come up dirty if there are untracked files in there - only if there are uncommitted  changes to tracked files